### PR TITLE
fix(rpc): return Err when client actor has detected syncing

### LINKED ISSUE #3492: Return non 200 for health rpc endpoint when the node is syncing
It suffices to return an error in the handling of `Status` message in `ClientActor` when `is_health_check` is true. This will greatly help with rpc load balancer to auto remove nodes that should not serve rpc.

### PR DESCRIPTION
**Pull request by ngmiller**
Resolves #3492

Return an `Err` Result when the client actor is syncing. This will help load balancers know when RPC traffic should not be routed to the node until it is caught up. (For new nodes syncing from a snapshot, or ones that fall behind for any reason.)

### CODE REVIEW

**Review by frol — APPROVED** [review]
Beware of #3059: 

> A node can be in syncing status as long as it is two blocks behind and this might happen just due to some network glitch. We can probably implement some sophisticated strategy for more fine-grained control, but for now it's better to revert it.

This may lead to nodes reporting they are not healthy and thus getting excluded from the balancing pool. Proceed with caution.

File: chain/client/src/client_actor.rs [hunk 1]
```diff
@@ -542,6 +542,10 @@ impl Handler<Status> for ClientActor {
                     return Err(format!("No blocks for {:?}.", elapsed));
                 }
             }
+
+            if self.client.sync_status.is_syncing() {
+                return Err(format!("Not healthy. Syncing!"));
```
Comment (bowenwang1996) [inline chain/client/src/client_actor.rs:547]: when there is nothing to format it is better to use `.to_string()`

**Review by bowenwang1996 — APPROVED** [review]

**Comment by ngmiller** [comment]
@bowenwang1996 @chefsale @mhalambek - I am a bit concerned about the point that @frol raised. One path of mitigation is to increase the "unhealthy" threshold on the GCP load-balancer, so 3 (or more) checks need to fail before it will be removed from the serving pool.

Do we think that is enough?

**Comment by bowenwang1996** [comment]
@ngmiller I think so. If a node is healthy and it suddenly fails the check 3 times, something is not right and the node should be removed.

**Comment by ngmiller** [comment]
@bowenwang1996 @mhalambek @chefsale This is ready

File: chain/client/src/client_actor.rs [hunk 2]
```diff
@@ -543,6 +543,10 @@ impl Handler<Status> for ClientActor {
                     return Err(format!("No blocks for {:?}.", elapsed));
                 }
             }
+
+            if self.client.sync_status.is_syncing() {
+                return Err("Not healthy. Syncing!".to_string());
```
Comment (SkidanovAlex) [inline chain/client/src/client_actor.rs:548]: Can it just be "The node is syncing", without "Not healthy" and the exclamation mark?

Comment (ngmiller) [reply chain/client/src/client_actor.rs]: done

**Pull request merged**